js/Utility.html

```diff
@@ -12,6 +12,77 @@
   array[index] = array[index].replace(/\s/g, '');
 },
 
+utility.stripItem = (value, index, array) => {
+  // only the quantity and barcode are needed, because the integrity
+  // check will pull the other information from the cache
+  array[index] = {
+    id: array[index].id,
+    description: array[index].description,
+    quantity: array[index].quantity,
+    barcode: array[index].barcode,
+  };
+},
+
+utility.itemIntegrityCheck = (item, index, array) => {
+  const items = app.cache.items;
+  const manualItemIdregex = /^MANUAL/;
+  var found = false;
+  // tries to locate the item in the inventory and
+  // sets the properties as they are in the inventory
+  for (let entry of items){
+    if (entry.barcode == item.barcode){
+      found = true;
+      item.description = entry.description;
+      item.serialized = entry.serialized;
+      if (item.serialized){
+        item.quantity = 1;
+      }
+      item.id = entry.id;
```

Comment:
I think you forgot a `break` here for when we found something... unless we need to find stuff multiple times?

Why not `array.prototype.find` (see above)?

Reply:
added